lib/src/main/kotlin/starknet/data/types/transactions.kt

```diff
@@ -79,12 +82,44 @@ class BlockHashOrTagSerializer() : KSerializer<BlockHashOrTag> {
 
 @Serializable
 data class InvokeFunctionPayload(
-    @SerialName("function_invocation") val invocation: Call,
+    @SerialName("function_invocation")
+    val invocation: Call,
+
     val signature: Signature?,
-    @SerialName("max_fee") val maxFee: Felt?,
+
+    @SerialName("max_fee")
+    val maxFee: Felt?,
+
     val version: Felt?
 )
 
+data class ContractDefinition(
+    val contract: String
+) {
+    fun parseContract(): Triple<JsonElement, JsonElement, JsonElement> {
+        val compiledContract = Json.parseToJsonElement(this.contract).jsonObject
+        val program = compiledContract["program"]!!
+        val entryPointsByType = compiledContract["entry_points_by_type"]!!
+        val abi = compiledContract["abi"] ?: JsonArray(emptyList())
+        return Triple(program, entryPointsByType, abi)
```

Comment:
Use data class instead or even better - make this a constructor for `ContractDefinition` with those fields (don't have to be public).